### PR TITLE
Search contract

### PR DESCRIPTION
**Pull request by Ic3Sandy**

### CODE REVIEW

File: docker-compose.fvm.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ services:
     platform: linux/amd64
     command: npm run node
     tty: true
+    restart: always
```
Comment (thebevrishot) [inline docker-compose.fvm.yml:18]: Let's do it just on productions for now we have to make sure this is really crash

File: packages/bitcore-node/src/modules/firocoin/models/contract.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  }
+  allowedPaging = [];
+
+  onConnect() {
+    this.collection.createIndex({ chain: 1, network: 1 }, { background: true });
+    this.collection.createIndex({ chain: 1, network: 1, contractAddress: 1 }, { background: true });
+    this.collection.createIndex({ txid: 1 }, { background: true });
+    this.collection.createIndex({ contractAddress: 1 }, { background: true });
+    this.collection.createIndex({ from: 1 }, { background: true });
+  }
```
Comment (thebevrishot) [inline packages/bitcore-node/src/modules/firocoin/models/contract.ts:46]: If there are unused indexes, please remove them.

File: packages/bitcore-node/src/modules/firocoin/models/contract.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        id: Date.now().toString(),
+        jsonrpc: '2.0',
+        method: 'gettransactionreceipt',
+        params: [txid]
+      })
+    };
```
Comment (thebevrishot) [inline packages/bitcore-node/src/modules/firocoin/models/contract.ts]: I think Bitcore should provide RPC wrapping.

File: packages/bitcore-node/src/modules/firocoin/p2p.ts [hunk 4]
```diff
@@ -242,6 +243,9 @@ export class FIROP2PWorker extends BaseP2PWorker<IBtcBlock> {
       initialSyncComplete: this.initialSyncComplete,
       block
     });
+    block.transactions.forEach(async tx => {
+      await ContractStorage.processContract({ chain: this.chain, network: this.network, txid: tx.hash });
+    });
```
Comment (thebevrishot) [inline packages/bitcore-node/src/modules/firocoin/p2p.ts]: do we need to await on this line? It's weird why RPC server could not handle 2 calls as a time

File: docker-compose.fvm.yml (hunk 1 above)
Comment (Ic3Sandy) [reply docker-compose.fvm.yml:18]: ok kub

File: packages/bitcore-node/src/modules/firocoin/models/contract.ts (hunk 2 above)
Comment (Ic3Sandy) [reply packages/bitcore-node/src/modules/firocoin/models/contract.ts:46]: ok kub

File: packages/bitcore-node/src/modules/firocoin/models/contract.ts (hunk 3 above)
Comment (Ic3Sandy) [reply packages/bitcore-node/src/modules/firocoin/models/contract.ts]: i will check kub

File: packages/bitcore-node/src/modules/firocoin/p2p.ts (hunk 4 above)
Comment (Ic3Sandy) [reply packages/bitcore-node/src/modules/firocoin/p2p.ts]: ok kub

**Pull request closed without merging**